numba/pycc/cc.py

```diff
@@ -272,7 +272,7 @@ class _CCExtension(Extension):
 
     def _prepare_object_files(self, build_ext):
         cc = self._cc
-        dir_util.mkpath(build_ext.build_temp)
+        dir_util.mkpath(os.path.join(build_ext.build_temp, *self.name.split('.')[:-1]))
```

Comment:
What is this call for?

Reply:
The `.o` files are created in folders as nested as modules, without that `gcc` fails.